### PR TITLE
Add default kernel config option

### LINKED ISSUE #620: Setup a default kernel based on user
/cc @glyph who has a bunch of custom kernels and not the magical Python3 default kernel we expect.

### LINKED ISSUE #4608: Add an option to configure a default runtime
This is about the windows desktop app.

I'm a julia user, and use nteract only for julia. It would be great if I could make the julia kernel my default kernel, so that any new notebook I create on my machine will use that kernel and I don't have to manually switch things over to julia for every new notebook I create.

### PR DESCRIPTION
**Pull request by CrystallineCat**
Fixes #4608. Fixes #620.

### CODE REVIEW

File: applications/desktop/src/main/index.ts [hunk 1]
```diff
@@ -274,16 +282,19 @@ openFile$
...
+        const defaultKernel = CONFIG.defaultKernel;
 
-        if (argv.kernel in specs) {
-          kernel = argv.kernel;
-        } else if ("python2" in specs) {
-          kernel = "python2";
+        if (passedKernel && passedKernel in specs) {
+          kernel = passedKernel;
+        } else if (defaultKernel && defaultKernel in specs) {
+          kernel = defaultKernel;
```
Comment (rgbkrk) [inline applications/desktop/src/main/index.ts:291]: Seems like a decent fallback here.

File: applications/desktop/src/main/menu.ts [hunk 2]
```diff
@@ -542,6 +547,15 @@ export function loadFullMenu(store = global.store) {
       {
         label: "Editor options",
         submenu: blink_menu
+      },
+      {
+        label: "Default kernel",
```
Comment (rgbkrk) [inline applications/desktop/src/main/menu.ts]: I'm a little curious about making this "Set default kernel" as the label. That's not a blocker on merging though.

File: applications/desktop/src/notebook/menu.ts [hunk 3]
```diff
@@ -555,8 +561,8 @@ export function dispatchNewNotebook(
   store: DesktopStore,
   event: Event,
   filepath: string | null,
-  kernelSpec: KernelSpec
-) {
+  kernelSpec?: KernelSpec,
```
Comment (rgbkrk) [inline applications/desktop/src/notebook/menu.ts]: What makes this optional now?

File: applications/desktop/src/main/menu.ts (hunk 2 above)
Comment (CrystallineCat) [reply applications/desktop/src/main/menu.ts]: Yeah, that might be a better label. I changed it.

File: applications/desktop/src/notebook/menu.ts (hunk 3 above)
Comment (CrystallineCat) [reply applications/desktop/src/notebook/menu.ts]: Thanks for catching that! I tried putting the default kernel resolution in the reducer, with the caller passing undefined unless a specific kernel was requested at the command line. This is a holdover from that. I removed the question mark.

**Comment by rgbkrk** [comment]
I think this looks good. There's a merge conflict you'll need to resolve in order to merge this. 😬

**Comment by CrystallineCat** [comment]
Thank you! 😀

One day, I will successfully fix a merge conflict without needing to hard reset and force push... One day! 🤕

**Comment by rgbkrk** [comment]
Force push is the only way to remedy it on your branch, it's a-ok. You're making new commits that match the updated history. 😄

**Pull request merged**